services/src/main/java/org/keycloak/services/ServicesLogger.java

```diff
@@ -467,4 +467,9 @@ public interface ServicesLogger extends BasicLogger {
     @Message(id=106, value="Created script engine '%s', version '%s' for the mime type '%s'")
     @Once
     void scriptEngineCreated(String engineName, String engineVersion, String mimeType);
+
+    @LogMessage(level = ERROR)
+    @Message(id=107, value="User '%s' in realm '%s' is not assigned %s role")
+    void userNotAssignedRole(String user, String realm, String role);
```

Comment:
@sjweixel Thanks for the contribution!

I wonder why we need this message. The error is that the user exists, not that it is not assigned an admin role.

I would maybe suggest a different approach for this PR. We could create a separate log message as suggested by @mabartos (something like `addAdminUserFailedAdminExists` maybe) and use it just in `KeycloakMain` without modifying `ApplienceBootstrap`.

Reply:
Thanks for updating the PR! I believe the comment above is still valid.

Reply:
I'm not clear on the comment.  The code in ApplianceBootStrap is testing if the user exists and if so, determines if the admin role is assigned to the user.

```
        if (existingUser != null) {
            RoleModel adminRole = realm.getRole(AdminRoles.ADMIN);
            if (existingUser.hasRole(adminRole)) {
               ServicesLogger.LOGGER.addAdminUserUserExists(username, Config.getAdminRealm());
            } else {
               ServicesLogger.LOGGER.userNotAssignedRole(username, Config.getAdminRealm(), AdminRoles.ADMIN);
            }
            return;
        }
```
Is your comment addressing one of these concerns?
- The log message about the role doesn't match the logic in the code.
- The change doesn't match the issue as it is described.
- This message has no value.

The initial issue that I opened was #16961.  It was closed as a duplicate of this issue, then reopened after my initial PR.  One of my concerns is that when KEYCLOAK_ADMIN is used to specify a user, an admin user isn't created if that username already exists without admin role.  This may not be a situation that occurs often.   I think it could the user some headache if the situation arises.

Reply:
@vmuzikar.  Please explain your comment about changing code in KeycloakMain instead of ApplianceBootstrap.  I'm not certhan that what you are suggesting is possible.  The code that currently checks for existing users and generates error log is in ApplianceBootstrap.  If user exists check and log message is added to KeycloakMain, then ApplianceBootrap will need also need to be modified to remove existing user check and log message.